### DISCUSSION TITLE
Add detailed logging to image uploads and other settings that require 3rd party connections

### DISCUSSION THREAD

**Discussion by fluteds** [Chatterino/chatterino2#4095] [Ideas] [answered]
> Moving discussion to here so it can be kept track of better. [Originally asked on Chatterino Discord.](https://discord.com/channels/253897423439659008/459650160658481152/1032689932721991753)

Like mod chat logging, 3rd party extensions should have an optional detailed log. E.g. for image uploading or other 3rd party connections, including expanded output for error logging. 

Ideally this should be added for troubleshooting what went wrong with the 3rd party service as at the moment logging is limited to chat input with no option to save logs. For example, I'm using https://sxcu.net as an image uploader with a token. The expanded error information shows an additional error code and description. 

Detailed output when sending a POST request.

```json
{
    "error": "Invalid upload token",
    "code": 809
}
```

Chatterino output on failing to send an image in chat.

```txt
12:18 Started upload... 
12:18 Uploading image: C:/Users/xxxx/image.jpg 
12:18 An error happened while uploading your image: 400 
```
Suggested chat logging:

```txt
12:18 Started upload... 
12:18 Uploading image: C:/Users/xxxx/image.jpg 
12:18 An error happened while uploading your image: code: 809  | error: Invalid upload token
```

**Comment by pajlada** [comment, marked as answer]
Solved in Nightly after abb69f679412153fdbc8de1918e11749d4cafd7f

@fluteds Could you try this PR to see if it works as expected? https://github.com/Chatterino/chatterino2/pull/4096

Either build it from source or wait for CI to build and download one of the artifacts from the "Checks" page on the PR

**Reply by fluteds** [reply]
@pajlada I used the Windows artifact build and can say it works exactly as intended. Thank you!
